lib/fluent/plugin_helper/socket.rb

```diff
@@ -150,8 +150,19 @@ def socket_create_tls(
           context.ciphers = ciphers
           context.verify_mode = OpenSSL::SSL::VERIFY_PEER
           context.cert_store = cert_store
-          context.verify_hostname = true if verify_fqdn && fqdn && context.respond_to?(:verify_hostname=)
-          context.cert = OpenSSL::X509::Certificate.new(File.read(cert_path)) if cert_path
+          if verify_fqdn && fqdn && context.respond_to?(:verify_hostname=)
+            context.verify_hostname = true
+          else
+            context.verify_hostname = false
```

Comment:
Why change this line to set `false` explicitly?

Reply:
see commit body of the first [commit](https://github.com/fluent/fluentd/pull/2700/commits/18d6846c046cbd081142444e3a5c0b3342fa3acd):
> Otherwise it seems to default to true, which results in logging of
>
>    warning: verify_hostname requires hostname to be set
>
> When running tests that use TLS without verifying the hostname.